src/main/kotlin/no/nav/familie/tilbake/dokumentbestilling/felles/BrevmetadataUtil.kt

```diff
@@ -0,0 +1,142 @@
+package no.nav.familie.tilbake.dokumentbestilling.felles
+
+import no.nav.familie.tilbake.behandling.BehandlingRepository
+import no.nav.familie.tilbake.behandling.BehandlingService
+import no.nav.familie.tilbake.behandling.FagsakRepository
+import no.nav.familie.tilbake.behandling.domain.Behandling
+import no.nav.familie.tilbake.behandling.domain.Fagsak
+import no.nav.familie.tilbake.behandlingskontroll.domain.Behandlingssteg
+import no.nav.familie.tilbake.common.repository.findByIdOrThrow
+import no.nav.familie.tilbake.config.Constants
+import no.nav.familie.tilbake.config.FeatureToggleConfig
+import no.nav.familie.tilbake.config.FeatureToggleService
+import no.nav.familie.tilbake.dokumentbestilling.DistribusjonshåndteringService
+import no.nav.familie.tilbake.dokumentbestilling.manuell.brevmottaker.ManuellBrevmottakerRepository
+import no.nav.familie.tilbake.dokumentbestilling.manuellAdresse
+import no.nav.familie.tilbake.dokumentbestilling.somBrevmottager
+import no.nav.familie.tilbake.dokumentbestilling.vedtak.Vedtaksbrevgrunnlag
+import no.nav.familie.tilbake.integration.pdl.internal.Personinfo
+import no.nav.familie.tilbake.organisasjon.OrganisasjonService
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.util.UUID
+
+@Service
+class BrevmetadataUtil(
+    private val behandlingRepository: BehandlingRepository,
+    private val fagsakRepository: FagsakRepository,
+    private val manuelleBrevmottakerRepository: ManuellBrevmottakerRepository,
+    private val eksterneDataForBrevService: EksterneDataForBrevService,
+    private val organisasjonService: OrganisasjonService,
+    private val featureToggleService: FeatureToggleService
+) {
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    fun genererMetadataForBrev(
+        behandlingId: UUID,
+        vedtaksbrevgrunnlag: Vedtaksbrevgrunnlag? = null,
+        brevmottager: Brevmottager = Brevmottager.MANUELL,
+        manuellAdresseinfo: Adresseinfo? = null,
+        annenMottakersNavn: String? = null,
+    ): Brevmetadata? {
+        require(brevmottager != Brevmottager.MANUELL || manuellAdresseinfo != null)
```

Comment:
Hva slags feilmelding kaster egentlig require her?
Jeg lurer på om man bør slenge på en egen feilmelding slik at det er forståelig for utviklere som kommer over dette senere på hvorfor dette er krav

Reply:
Sikkert ingenting forklarende.. Kan godt legge på en feilmelding 👍